### PR TITLE
[core][c++ worker]Add namespace support for c++ worker

### PR DESCRIPTION
**Pull request by WangTaoTheTonic**
## Why are these changes needed?
Namespace feature [has been supported in Python/Java](https://docs.ray.io/en/latest/ray-core/namespaces.html) but not in c++ worker. All created actors are assigned with a default empty namespace.

With this change, users can use namespace in c++ worker like:
```
RayConfig config;
config.ray_namespace = "my_ns";
ray::Init(config, ..., ...);
```
Then all named actor created are under namespace `my_ns`, they're invisible to other namespaces.
```
auto test_actor = ray::Actor(...).setName("name", "test_ns").Remote();
``` 
Then `test_actor` is under namespace `test_ns`, it's only visible to those who is under same one.
User can also use api with namespace to get an actor:
```
ray::GetActor<...>(actor_name, namespace_name);
```

- [x] ray::GetActor with namespace
- [x] some more test cases

I'll file another PR to update the doc.

## Related issue number

https://github.com/ray-project/ray/issues/26183
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by jovany-wang** [comment]
FYI: We are aiming to remove job config concept, see https://github.com/ray-project/ray/pull/26116
but I think we could merge this PR firstly, and then remove job config in C++ worker after that PR merged.

File: cpp/include/ray/api/ray_config.h [hunk 1]
```diff
@@ -59,6 +59,8 @@ class RayConfig {
 
   // A specific flag for internal `default_worker`. Please don't use it in user code.
   bool is_worker_ = false;
+
```
Comment (jovany-wang) [inline cpp/include/ray/api/ray_config.h:62]: Comment on the field?

File: src/ray/util/util.h [hunk 2]
```diff
@@ -117,6 +117,38 @@ inline int64_t current_sys_time_us() {
   return mu_since_epoch.count();
 }
 
+static std::random_device              rd;
+static std::mt19937                    gen(rd());
+static std::uniform_int_distribution<> dis(0, 15);
+static std::uniform_int_distribution<> dis2(8, 11);
+
+inline std::string generate_uuid_v4() {
```
Comment (jovany-wang) [inline src/ray/util/util.h:125]: According to our code style, it should be:
```suggestion
inline std::string GenerateUUIDV4() {
```

Comment (jovany-wang) [reply src/ray/util/util.h:125]: According to C++ practices, we should avoid global static variables as possible. I think there are 2 options to make the code graceful:
1. Add a UUIDV4Generator class and then make `random_device` `mt19937`  as class static variables.
2. make them as local static variable, like:
```C++
inline std::string generate_uuid_v4() {
    static std::random_device              rd;
    static std::mt19937                    gen(rd());
    static std::uniform_int_distribution<> dis(0, 15);
    static std::uniform_int_distribution<> dis2(8, 11);

    std::stringstream ss;
      int i;
      ss << std::hex;
      for (i = 0; i < 8; i++) {
        ss << dis(gen);
      }
    }
```

**Review by jovany-wang — COMMENTED** [review]
Nice PR, I left some minor comments related code style and practices.

File: src/ray/util/util.h [hunk 3]
```diff
@@ -117,6 +117,38 @@ inline int64_t current_sys_time_us() {
   return mu_since_epoch.count();
 }
 
+static std::random_device              rd;
```
Comment (raulchen) [inline src/ray/util/util.h:120]:
```suggestion
static std::random_device rd;
```
IIRC, our lint rule should prefer this style. Right?

**Review by raulchen — APPROVED** [review]

File: src/ray/util/util.h (hunk 3 above)
Comment (WangTaoTheTonic) [reply src/ray/util/util.h:120]: yeah, it's been detected by lint. I'll change it.

File: cpp/include/ray/api.h [hunk 4]
```diff
@@ -120,6 +120,18 @@ ray::internal::ActorCreator<JavaActorClass> Actor(JavaActorClass func);
 template <typename T>
 boost::optional<ActorHandle<T>> GetActor(const std::string &actor_name);
 
+/// Get a handle to a named actor in the given namespace.
+/// Gets a handle to a named actor with the given name of the given namespace. The actor
+/// must have been created with name specified.
```
[the remark this answers is not in the document]
Comment (WangTaoTheTonic) [reply cpp/include/ray/api.h:125]: Yeah, it's copied from the upper line, I'll change them both.

File: src/ray/util/util.h (hunk 2 above)
Comment (WangTaoTheTonic) [reply src/ray/util/util.h:125]: Yeah, I'd prefer the second one. What do you think?

Comment (jovany-wang) [reply src/ray/util/util.h:125]: Yes. the 1st one is a bit heavy.

File: cpp/include/ray/api/ray_config.h (hunk 1 above)
Comment (WangTaoTheTonic) [reply cpp/include/ray/api/ray_config.h:62]: ok

**Review by jovany-wang — APPROVED** [review]

**Pull request merged**